File: gamslib/jobt/jobt-scalar.py
#  LP written by GAMS Convert at 12/13/18 10:24:28
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         16       11        5        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         31       31        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         73       73        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x5 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x10 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x11 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x12 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x17 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x18 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x19 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x20 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x21 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x22 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x23 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x24 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x25 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x26 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x27 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x28 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x29 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x30 = Var(within=Reals,bounds=(0,None),initialize=0)

m.obj = Objective(expr=   10*m.x6 + 10*m.x7 + 10*m.x8 + 10*m.x9 + 10*m.x10 + 30*m.x11 + 30*m.x12 + 30*m.x13 + 30*m.x14
                        + 30*m.x15 + 100*m.x16 + 100*m.x17 + 100*m.x18 + 100*m.x19 + 200*m.x20, sense=minimize)

m.c1 = Constraint(expr= - m.x1 + m.x6 - m.x11 == -90)

m.c2 = Constraint(expr= - m.x2 - m.x6 + m.x7 + m.x11 - m.x12 == -200)

m.c3 = Constraint(expr= - m.x3 - m.x7 + m.x8 + m.x12 - m.x13 == -300)

m.c4 = Constraint(expr= - m.x4 - m.x8 + m.x9 + m.x13 - m.x14 == -400)

m.c5 = Constraint(expr= - m.x5 - m.x9 + m.x10 + m.x14 - m.x15 == -200)

m.c6 = Constraint(expr=   m.x16 - m.x21 + m.x26 == 20)

m.c7 = Constraint(expr= - m.x16 + m.x17 - m.x22 + m.x27 == 0)

m.c8 = Constraint(expr= - m.x17 + m.x18 - m.x23 + m.x28 == 0)

m.c9 = Constraint(expr= - m.x18 + m.x19 - m.x24 + m.x29 == 0)

m.c10 = Constraint(expr= - m.x19 + m.x20 - m.x25 + m.x30 == 0)

m.c11 = Constraint(expr= - 0.125*m.x1 + m.x16 - 1.16666666666667*m.x21 >= 0)

m.c12 = Constraint(expr= - 0.125*m.x2 + m.x17 - 1.16666666666667*m.x22 >= 0)

m.c13 = Constraint(expr= - 0.125*m.x3 + m.x18 - 1.16666666666667*m.x23 >= 0)

m.c14 = Constraint(expr= - 0.125*m.x4 + m.x19 - 1.16666666666667*m.x24 >= 0)

m.c15 = Constraint(expr= - 0.125*m.x5 + m.x20 - 1.16666666666667*m.x25 >= 0)

File: gamslib/jobt/jobt-scalar.jl
#  LP written by GAMS Convert at 10/11/20 12:15:38
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         16       11        5        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         31       31        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         73       73        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x5, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)
@variable(m, 0 <= x10, start=0)
@variable(m, 0 <= x11, start=0)
@variable(m, 0 <= x12, start=0)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= x17, start=0)
@variable(m, 0 <= x18, start=0)
@variable(m, 0 <= x19, start=0)
@variable(m, 0 <= x20, start=0)
@variable(m, 0 <= x21, start=0)
@variable(m, 0 <= x22, start=0)
@variable(m, 0 <= x23, start=0)
@variable(m, 0 <= x24, start=0)
@variable(m, 0 <= x25, start=0)
@variable(m, 0 <= x26, start=0)
@variable(m, 0 <= x27, start=0)
@variable(m, 0 <= x28, start=0)
@variable(m, 0 <= x29, start=0)
@variable(m, 0 <= x30, start=0)

@objective(m, Min, 10*x6 + 10*x7 + 10*x8 + 10*x9 + 10*x10 + 30*x11 + 30*x12 + 30*x13 + 30*x14 + 30*x15 + 100*x16
     + 100*x17 + 100*x18 + 100*x19 + 200*x20)

@constraint(m,  - x1 + x6 - x11 == -90)

@constraint(m,  - x2 - x6 + x7 + x11 - x12 == -200)

@constraint(m,  - x3 - x7 + x8 + x12 - x13 == -300)

@constraint(m,  - x4 - x8 + x9 + x13 - x14 == -400)

@constraint(m,  - x5 - x9 + x10 + x14 - x15 == -200)

@constraint(m, x16 - x21 + x26 == 20)

@constraint(m,  - x16 + x17 - x22 + x27 == 0)

@constraint(m,  - x17 + x18 - x23 + x28 == 0)

@constraint(m,  - x18 + x19 - x24 + x29 == 0)

@constraint(m,  - x19 + x20 - x25 + x30 == 0)

@constraint(m,  - 0.125*x1 + x16 - 1.16666666666667*x21 >= 0)

@constraint(m,  - 0.125*x2 + x17 - 1.16666666666667*x22 >= 0)

@constraint(m,  - 0.125*x3 + x18 - 1.16666666666667*x23 >= 0)

@constraint(m,  - 0.125*x4 + x19 - 1.16666666666667*x24 >= 0)

@constraint(m,  - 0.125*x5 + x20 - 1.16666666666667*x25 >= 0)
